js/__internal/grids/grid_core/data_controller/module.ts

```diff
@@ -36,1351 +88,1496 @@ export const dataControllerModule = {
+          } else {
+            that.updateItems({ repaintChangesOnly: options.changesOnly });
+            d.resolve();
           }
-        },
+        });
 
-        skipProcessingPagingChange(fullName) {
-          return this._skipProcessingPagingChange && (fullName === 'paging.pageIndex' || fullName === 'paging.pageSize');
-        },
+        return d.promise();
+      }
 
-        getUserState() {
-          return {
-            searchText: this.option('searchPanel.text'),
-            pageIndex: this.pageIndex(),
-            pageSize: this.pageSize(),
-          };
-        },
-
-        getCachedStoreData() {
-          return this._dataSource && this._dataSource.getCachedStoreData();
-        },
-
-        isLastPageLoaded() {
-          const pageIndex = this.pageIndex();
-          const pageCount = this.pageCount();
-          return pageIndex === (pageCount - 1);
-        },
-      };
-
-      gridCoreUtils.proxyMethod(members, 'load');
-      gridCoreUtils.proxyMethod(members, 'reload');
-      gridCoreUtils.proxyMethod(members, 'push');
-      gridCoreUtils.proxyMethod(members, 'itemsCount', 0);
-      gridCoreUtils.proxyMethod(members, 'totalItemsCount', 0);
-      gridCoreUtils.proxyMethod(members, 'hasKnownLastPage', true);
-      gridCoreUtils.proxyMethod(members, 'isLoaded', true);
-      gridCoreUtils.proxyMethod(members, 'totalCount', 0);
-
-      return members;
-    })()),
+      getVisibleRows() {
+        return this.items();
+      }
+
+      _disposeDataSource() {
+        this.setDataSource(null);
+      }
+
+      dispose() {
+        this._disposeDataSource();
+        super.dispose();
+      }
+
+      repaintRows(rowIndexes, changesOnly) {
+        rowIndexes = Array.isArray(rowIndexes) ? rowIndexes : [rowIndexes];
+
+        if (rowIndexes.length > 1 || isDefined(rowIndexes[0])) {
+          this.updateItems({ changeType: 'update', rowIndices: rowIndexes, isFullUpdate: !changesOnly });
+        }
+      }
+
+      skipProcessingPagingChange(fullName) {
+        return this._skipProcessingPagingChange && (fullName === 'paging.pageIndex' || fullName === 'paging.pageSize');
+      }
+
+      getUserState() {
+        return {
+          searchText: this.option('searchPanel.text'),
+          pageIndex: this.pageIndex(),
+          pageSize: this.pageSize(),
+        };
+      }
+
+      getCachedStoreData() {
+        return this._dataSource && this._dataSource.getCachedStoreData();
+      }
+
+      isLastPageLoaded() {
+        const pageIndex = this.pageIndex();
+        const pageCount = this.pageCount();
+        return pageIndex === (pageCount - 1);
+      }
+
+      load(): any {
+        return this._dataSource?.load();
+      }
+
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      reload(reload?, changesOnly?): any {
+        return this._dataSource?.reload(reload, changesOnly);
+      }
+
+      push() {
+        return this._dataSource?.push();
+      }
```

Comment:
Let's add `...arguments` to all proxied methods
```suggestion
      push() {
        return this._dataSource?.push(...arguments);
      }
```